### DISCUSSION TITLE
Sort Posts in Publii 0.46 with Mercury Theme

### DISCUSSION THREAD

**Discussion by eyefollowu** [GetPublii/Publii#1747] [Help] [answered]
I'm just starting with Publii 0.46 and the mercury theme to build my new photography portfolio site. Currently I'm very pleased with it and it nearly does everything I need. I'd like to display my several projets as posts, but I can't find any option to sort the posts to my liking. Is there any possibility to do so? Could Publii be extended by a plugin or some javascript? Could someone provide such a script? I assume many users with nearly to zero development skills like me would appreciate such a function in Publii. Thank you in advance.

**Comment by candidexmedia** [comment]
What specific type of sort are you looking for?

**Comment by michaelfluegge** [comment]
Try Site settings -> Advanced options -> Post listing -> Posts order by: .
You can choose by creation date, title, modification date, ID.  And you can order ask or desc. In addition you can deine some of your posts as Featured to stay at top, again asc or desc.
If that's not enough, I am sure, @candidexmedia will find a solution for you, e.g. by modifying the config.json.

**Comment by eyefollowu** [comment]
Thank you very much for the help. I just tried to change the settings for the Post listing, but unfortunately this has no effect when I preview the site in Firefox. The Post order remains the same, even if I mark some posts as featured. Then the featured posts are flagged, but the order the posts are displayed in the browser remains the same as before.

My wish would be to sort the posts fully manually, e.g. with drag und drop in the post listing. That would be intuitive and satisfying.  

If this isn't possible I assume that I could live by sorting the posts by modification date in the advanced options settings. I modified and saved some posts, but this has no effect. Maybe there ist a bug?

**Reply by bobmitro** [reply]
It is possible, but first, you must remove the default post loop and replace it with the `{{#getPosts}}` helper https://getpublii.com/dev/getposts-helper/.

**Reply by michaelfluegge** [reply]
@eyefollowu, I sometimes run into the error that I use the wrong preview button. It is important to use the buttons on the very top or or bottom of the app, not(!) on the bottom left (on the blue ribbon).

**Reply by eyefollowu** [reply]
Thank you, @bobmitro. I'll have a look at the getposts-helper and report my findings here.

And thx to you, @michaelfluegge. I just tried your suggestion, but this still has no effect on the order the broser displays the post. Even if I change the sorting of the posts in the app, e.g. from ID to modification date. The order of the posts changes in the app then, but this has no effect on the displayed website. Any other ideas?

**Reply by michaelfluegge** [reply]
If you are working with Safari for example, I would suggest to clear the cache ( CMD-R). I sometimes run into the problem, that Safari does not show changes until I refresh the page

**Reply by eyefollowu** [reply]
@michaelfluegge, thank you for the suggestion. I'm on Windows, my main browser is Firefox. The cache is cleared on exit, so this should not be the problem. I even tested with chromium and Edge and additionally on a second computer. The settings in Publii to sort the posts are displayed in the post listing of the app, but have no effect on the previewed website, unfortunately - regardless of the sort settings in Publii. On the previewed website the last created post stays on top and if I create a new one, the new post gets the top position. 
To me this looks like a bug. Anybody else with this problem?

@bobmitro, I had a short look at the {{#getPosts}} helper, but honestly at first sight I didn't understand much of it. Could you give a more detailed how-to for a nonprofessional like me? I'm just a photographer, no programmer... I would greatly appreciate any further help.

**Reply by michaelfluegge** [reply]
how about cloning the website (https://getpublii.com/docs/creating-your-website.html -> scroll down to "Duplicating a Website") and have a look whether sorting works then? And/Or changing the theme? Just for finding out where the problem might be located.
@bobmitro is usually very busy with developing the app so I would not count on an answer by him.

**Comment by eyefollowu** [comment]
@michaelfluegge, thanks again! I duplicated the site and after a test I additionally changed to the Simple theme. The problem remains, sadly.
For today I shut down the computer. I'll be busy the next days and will have a look again at it after that.

**Comment by eyefollowu** [comment, marked as answer]
After going through all the settings in Publii again I finally found the solution. In the 'app settings' under 'default lists ordering' the 'default posts ordering' is only responsible for the posts ordering displayed in Publii - this has nothing to do with the post ordering on the rendered website. 
To get the post orderd on the rendered website I have to go to 'Site Settings', then to 'Advanced Options / Posts Listing' and set the desired order there under 'Posts order by'. I set it to 'Post modification date' and can now order my posts by editing and saving the posts again in the desired order.

Thanks to everybody again and sorry for the confusion.

**Reply by michaelfluegge** [reply]
That was the tip I gave you last week, remember? ;-) Nevermind, I started that way, too...
